notification/store.go

```diff
@@ -266,41 +268,41 @@ func (db *DB) SendContactMethodVerification(ctx context.Context, id string, rese
 	return tx.Commit()
 }
 
-func (db *DB) VerifyContactMethod(ctx context.Context, cmID string, code int) ([]string, error) {
+func (db *DB) VerifyContactMethod(ctx context.Context, cmID string, _code string) error {
 	userID, err := db.cmUserID(ctx, cmID)
 	if err != nil {
-		return nil, err
+		return err
+	}
+
+	code, err := strconv.ParseInt(_code, 10, 64)
+	if err != nil {
+		return validation.NewFieldError("Code", "must contain only digits")
+	}
+
+	matched, _ := regexp.MatchString(`^\d{6}$`, _code)
+	if !matched {
+		return validation.NewFieldError("Code", "must be 6 digits")
 	}
 
 	tx, err := db.db.BeginTx(ctx, nil)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer tx.Rollback()
 
 	var cmValue string
 	err = db.verifyVerificationCode.QueryRowContext(ctx, cmID, code).Scan(&cmValue)
 	if err == sql.ErrNoRows {
-		return nil, validation.NewFieldError("Code", "unrecognized code")
+		return validation.NewFieldError("Code", "unrecognized code")
 	}
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	rows, err := db.enableContactMethods.QueryContext(ctx, userID, cmValue)
+	_, err = db.enableContactMethods.QueryContext(ctx, userID, cmValue)
```

Comment:
Let's update this to only enable the single `cmID` instead of all. That also means we can update the verify query to not return the contact method info.

Lastly, these queries should be executed in the transaction with `tx.StmtContext()`. Somehow we missed that before.